File: C++/Kata.LoanPrediction.C++/Kata.LoanPrediction.C++/LoanContext.cpp
//
//  LoanContext.cpp
//  Kata.LoanPrediction.C++
//
//  Created by retroburst [Andrew D] on 3/09/2015.
//  Copyright (c) 2015 retroburst [Andrew D]. All rights reserved.
//

#include "LoanContext.h"

/*
 ***************************************
 ** Default constructor.
 ***************************************
 */
LoanContext::LoanContext()
{ }

/*
 ***************************************
 ** Constructor.
 ***************************************
 */
LoanContext::LoanContext(dateTime startDate, dateTime targetEndDate, dateTime todaysDate,
                         double balance, double interestRate, double minRepaymentAmount,
                         int minRepaymentDay, double extraRepaymentAmount, int extraRepaymentDay)
{
    this->startDate = startDate;
    this->targetEndDate = targetEndDate;
    this->todaysDate = todaysDate;
    this->balance = balance;
    this->interestRate = interestRate;
    this->minRepaymentAmount = minRepaymentAmount;
    this->minRepaymentDay = minRepaymentDay;
    this->extraRepaymentAmount = extraRepaymentAmount;
    this->extraRepaymentDay = extraRepaymentDay;
}

/*
 ***************************************
 ** Get start date.
 ***************************************
 */
dateTime LoanContext::getStartDate()
{
    return(startDate);
}

/*
 ***************************************
 ** Set start date.
 ***************************************
 */
void LoanContext::setStartDate(dateTime source)
{
    startDate = source;
}

/*
 ***************************************
 ** Get target end date.
 ***************************************
 */
dateTime LoanContext::getTargetEndDate()
{
    return(targetEndDate);
}

/*
 ***************************************
 ** Set target end date.
 ***************************************
 */
void LoanContext::setTargetEndDate(dateTime source)
{
    targetEndDate = source;
}

/*
 ***************************************
 ** Get today's date.
 ***************************************
 */
dateTime LoanContext::getTodaysDate()
{
    return(todaysDate);
}

/*
 ***************************************
 ** Set today's date.
 ***************************************
 */
void LoanContext::setTodaysDate(dateTime source)
{
    todaysDate = source;
}

/*
 ***************************************
 ** Get balance.
 ***************************************
 */
double LoanContext::getBalance()
{
    return(balance);
}

/*
 ***************************************
 ** Set balance.
 ***************************************
 */
void LoanContext::setBalance(double source)
{
    balance = source;
}

/*
 ***************************************
 ** Get interest rate.
 ***************************************
 */
double LoanContext::getInterestRate()
{
    return(interestRate);
}

/*
 ***************************************
 ** Set interest rate.
 ***************************************
 */
void LoanContext::setInterestRate(double source)
{
    interestRate = source;
}

/*
 ***************************************
 ** Get min. repayment amount.
 ***************************************
 */
double LoanContext::getMinRepaymentAmount()
{
    return(minRepaymentAmount);
}

/*
 ***************************************
 ** Set min. repayment amount.
 ***************************************
 */
void LoanContext::setMinRepaymentAmount(double source)
{
    minRepaymentAmount = source;
}

/*
 ***************************************
 ** Get min. repayment day.
 ***************************************
 */
int LoanContext::getMinRepaymentDay()
{
    return(minRepaymentDay);
}

/*
 ***************************************
 ** Set min. repayment day.
 ***************************************
 */
void LoanContext::setMinRepaymentDay(int source)
{
    minRepaymentDay = source;
}

/*
 ***************************************
 ** Get extra repayment amount.
 ***************************************
 */
double LoanContext::getExtraRepaymentAmount()
{
    return(extraRepaymentAmount);
}

/*
 ***************************************
 ** Set extra repayment amount.
 ***************************************
 */
void LoanContext::setExtraRepaymentAmount(double source)
{
    extraRepaymentAmount = source;
}

/*
 ***************************************
 ** Get extra repayment day.
 ***************************************
 */
int LoanContext::getExtraRepaymentDay()
{
    return(extraRepaymentDay);
}

/*
 ***************************************
 ** Set extra repayment day.
 ***************************************
 */
void LoanContext::setExtraRepaymentDay(int source)
{
    extraRepaymentDay = source;    
}

File: C#/Kata.LoanPrediction.CSharp/Kata.LoanPrediction.CSharp.Common/Models/LoanContext.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Kata.LoanPrediction.CSharp.Common.Models
{
    /// <summary>
    /// Represents the context of a loan.
    /// </summary>
    public class LoanContext
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="LoanContext"/> class.
        /// </summary>
        public LoanContext()
        { }

        /// <summary>
        /// Initializes a new instance of the <see cref="LoanContext"/> class.
        /// </summary>
        /// <param name="todaysDate">The todays date.</param>
        /// <param name="startDate">The start date.</param>
        /// <param name="targetEndDate">The target end date.</param>
        /// <param name="balance">The balance.</param>
        /// <param name="interestRate">The interest rate.</param>
        /// <param name="minRepaymentAmount">The minimum repayment amount.</param>
        /// <param name="minRepaymentDay">The minimum repayment day.</param>
        /// <param name="extraRepaymentAmount">The extra repayment amount.</param>
        /// <param name="extraRepaymentDay">The extra repayment day.</param>
        public LoanContext(DateTime todaysDate, DateTime startDate, DateTime targetEndDate, double balance,
            double interestRate, double minRepaymentAmount, int minRepaymentDay, double extraRepaymentAmount,
            int extraRepaymentDay)
        {
            TodaysDate = todaysDate;
            StartDate = startDate;
            TargetEndDate = targetEndDate;
            Balance = balance;
            InterestRate = interestRate;
            MinRepaymentAmount = minRepaymentAmount;
            MinRepaymentDay = minRepaymentDay;
            ExtraRepaymentAmount = extraRepaymentAmount;
            ExtraRepaymentDay = extraRepaymentDay;
        }

        /// <summary>
        /// Gets or sets the todays date.
        /// </summary>
        /// <value>
        /// The todays date.
        /// </value>
        public DateTime TodaysDate { get; set; }
        
        /// <summary>
        /// Gets or sets the start date.
        /// </summary>
        /// <value>
        /// The start date.
        /// </value>
        public DateTime StartDate { get; set; } 
        
        /// <summary>
        /// Gets or sets the target end date.
        /// </summary>
        /// <value>
        /// The target end date.
        /// </value>
        public DateTime TargetEndDate { get; set; } 
        
        /// <summary>
        /// Gets or sets the balance.
        /// </summary>
        /// <value>
        /// The balance.
        /// </value>
        public double Balance {get; set; } 
        
        /// <summary>
        /// Gets or sets the interest rate.
        /// </summary>
        /// <value>
        /// The interest rate.
        /// </value>
        public double InterestRate {get; set;} 
        
        /// <summary>
        /// Gets or sets the minimum repayment amount.
        /// </summary>
        /// <value>
        /// The minimum repayment amount.
        /// </value>
        public double MinRepaymentAmount {get;set;}
        
        /// <summary>
        /// Gets or sets the minimum repayment day.
        /// </summary>
        /// <value>
        /// The minimum repayment day.
        /// </value>
        public int MinRepaymentDay { get; set;} 
                
        /// <summary>
        /// Gets or sets the extra repayment amount.
        /// </summary>
        /// <value>
        /// The extra repayment amount.
        /// </value>
        public double ExtraRepaymentAmount { get; set;}
        
        /// <summary>
        /// Gets or sets the extra repayment day.
        /// </summary>
        /// <value>
        /// The extra repayment day.
        /// </value>
        public int ExtraRepaymentDay { get; set; }
    }
}
